src/browser-driver-manager.js

```diff
@@ -1,259 +1,284 @@
- * @param {Boolean} [options.sudo] - Run the command as sudo
- * @param {String[]} options.args - List of script arguments
- * @returns {Promise<String|null>}
+ * Install a version of Chrome and Chromedriver.
+ * If already-installed Chrome and Chromedriver match this version, skip installation.
+ * If already-installed Chrome and Chromedriver are a different version, overwrite them.
+ * @param {string} browserId - Browser name (chrome), with an optional `@` and version number (e.g. 116.0.5845.96) or channel (e.g. beta, dev, canary)
+ * @param {object} options - Pass-through to installBrowser
+ * @throws {Error} Browser must be chrome
+ * @throws {Error} Browser platform must be detectable
+ * @throws {Error} Build ID must resolve
+ * @throws {Error} Browser must be uninstallable when one is already installed
+ * @throws {Error} Version of chrome and chromedriver must be findable
+ * @throws {Error} Environment file must be removable when one already exists
+ * @returns
  */
-function runBashScript({ returnValue = false, sudo = false, args } = {}) {
-  return new Promise((resolve, reject) => {
-    const lines = [];
-    let lastChunk;
+async function install(browserId, options) {
+  const [browser, version = 'stable'] = browserId.split('@');
+
+  // Should support for other browsers be added, commander should handle this check.
+  // With only one supported browser, this error message is more meaningful than commander's.
+  if (!browser.includes('chrome')) {
+    throw new Error(
+      `The selected browser, ${browser}, could not be installed. Currently, only "chrome" is supported.`
+    );
+  }
 
-    const scriptPath = path.join(__dirname, 'index.sh');
-    let child;
-    if (sudo) {
-      child = spawn('sudo', [scriptPath, ...args]);
-    } else {
-      child = spawn(scriptPath, args);
+  const platform = detectBrowserPlatform();
+
+  if (!platform) {
+    throw new Error(
+      `Unable to detect a valid platform for ${process.platform} ${process.arch}. Darwin (MacOS, iOS, etc.), Windows, and Linux platforms are supported.`
+    );
+  }
+
+  // Get the version to install of both Chrome and Chromedriver
+  const buildId = await resolveBuildId(Browser.CHROME, platform, version);
+
+  // Get any currently installed version
+  const currentVersion = await getVersion(true);
+  const cacheDir = getBDMCacheDir();
+
+  if (currentVersion) {
+    if (currentVersion === buildId) {
+      console.log(
+        `Chrome and Chromedriver versions ${currentVersion} are already installed. Skipping installation.`
+      );
+      return;
     }
+    console.log(
+      `Chrome and Chromedriver versions ${currentVersion} are currently installed. Overwriting.`
+    );
+  }
 
-    child.stdout.on('data', chunk => {
-      lines.push(chunk);
+  const envPath = path.resolve(cacheDir, '.env');
+  try {
+    // Remove the existing .env, if it exists. setEnv creates it again later
```

Comment:
You'll also need to put the removing of the env after install otherwise you can still get into the problem of installing a valid version, installing an invalid version, and now `.env` doesn't exist so are in a bad state again.

I would suggest having a test or two that verifies this logic as well. You can have an `.env` file already exist then have the install mock return an error. If you still have the `.env` and it's the same as before the install it should pass.